### PR TITLE
Small improvements to t commands

### LINKED ISSUE #118: 'mssql' db type not supported by tdb
```tdb``` does not support the db_type ```mssql```, despite it being a valid db type in older versions of totara. Currently if you want to run mssql, you need to specify the db type as ```sqlsvr```. We should really change it to accept either ```mssql``` OR ```sqlsvr```.

### PR DESCRIPTION
**Pull request by derschatta**
fixes #118 and #117

### CODE REVIEW

**Review by markmetcalfe — CHANGES_REQUESTED** [review]

File: bin/tnpm [hunk 1]
```diff
@@ -9,4 +9,4 @@ export $(egrep -v '^#' $PROJECTPATH/.env | xargs)
 SUBPATH="${PWD//$LOCAL_SRC/}"
 TOTARASITEDIR="$REMOTE_SRC/$SUBPATH"
 
-$SCRIPTPATH/tdocker run -w "$TOTARASITEDIR" nodejs npm "$@"
+$SCRIPTPATH/tdocker run --rm -w "$TOTARASITEDIR" nodejs npm "$@"
```
[the remark this answers is not in the document]
Comment (derschatta) [reply bin/tnpm:12]: Without the --rm the command will leave a volume behind for each tui command you trigger. With the command it just cleans up the volume used for the one run.

File: bin/tdb [hunk 2]
```diff
@@ -373,7 +373,7 @@ elif [ "$db_type" == 'mariadb' ]; then
 ##########################################
 #          Microsoft SQL Server          #
 ##########################################
-elif [ "$db_type" == 'sqlsrv' ]; then
+elif [ "$db_type" == 'sqlsrv' -o  "$db_type" == 'mssql' ]; then
```
[the remark this answers is not in the document]
Comment (derschatta) [reply bin/tdb:376]: Just wanted to be consistent to the other conditions here and keep using single brackets, but happy to change it nevertheless

File: bin/tbash [hunk 3]
```diff
@@ -1,5 +1,17 @@
 #!/bin/bash
 
 SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+PROJECTPATH="$( cd $SCRIPTPATH && cd ..; pwd -P )"
 
-$SCRIPTPATH/tdocker exec "$1" /bin/bash
+export $(egrep -v '^#' $PROJECTPATH/.env | xargs)
+
+# If tnpm is called from a subdirectory, then we run npm in that subdirectory too
```
[the remark this answers is not in the document]
Comment (derschatta) [reply bin/tbash]: changed

File: bin/tdb (hunk 2 above)
[the remark this answers is not in the document]
Comment (derschatta) [reply bin/tdb]: changed

**Comment by derschatta** [comment]
this also tackles #110 now adding support for changing the backup path via the .env file and also removing the need of specifing full file paths/names. Now you just choose a name for the backup and tdb will take care of putting it into the right place and it will also attach the dbhost.

so `tdb backup totara_13 mybackup` will result in `~/tdb_backup/mybackup.pgsql11` for example.

File: bin/tnpm (hunk 1 above)
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply bin/tnpm:12]: Ah cool yeah its annoying how it did that

**Comment by derschatta** [comment]
@markmetcalfe if you are fine with the changes please approve the review

**Review by markmetcalfe — COMMENTED** [review]
> this also tackles #110 now, adding support for changing the backup path via the .env file and also removing the need of specifying full file paths/names. Now you just choose a name for the backup and tdb will take care of putting it into the right place and it will also attach the dbhost.
> 
> so `tdb backup totara_13 mybackup` will result in `~/tdb_backup/mybackup.pgsql11` for example.

I'm not a fan of restricting backups to only be able to be used in the same version. If its done like this, then you can't backup a postgres 11 database and then restore it into postgres 12 for instance. Instead I think it should be that you also specify the host when backing up restoring like
tdb backup totara13 bigdata.pgsql11
tdb restore totara13 bigdata.pgsql11

**Comment by derschatta** [comment]
> > this also tackles #110 now, adding support for changing the backup path via the .env file and also removing the need of specifying full file paths/names. Now you just choose a name for the backup and tdb will take care of putting it into the right place and it will also attach the dbhost.
> > so `tdb backup totara_13 mybackup` will result in `~/tdb_backup/mybackup.pgsql11` for example.
> 
> I'm not a fan of restricting backups to only be able to be used in the same version. If its done like this, then you can't backup a postgres 11 database and then restore it into postgres 12 for instance. Instead I think it should be that you also specify the host when backing up restoring like
> `tdb backup totara13 bigdata.pgsql11`
> `tdb restore totara13 bigdata.pgsql11`

what about a compromise and we use the database type instead, so it would become `totara13 bigdata.pgsql` and you could restore it into any db. The reason I did this automatic suffixing is to make it easier for the user and if he wants to use the same name for different databases he could do so without explicitely specifying them. Also having the database type in the name is really helpful if you have a lot of dumps.

**Comment by markmetcalfe** [comment]
> what about a compromise and we use the database type instead, so it would become `totara13 bigdata.pgsql` and you could restore it into any db of the same type? The reason I did this automatic suffixing is to make it easier for the user and if he wants to use the same name for different databases he could do so without explicitely specifying them. Also having the database type in the name is really helpful if you have a lot of dumps.
OK sure.

**Comment by derschatta** [comment]
@derschatta changed the backup file names to include the db type instead of host.

I also changed the default collation for mariadb/mysql as utf8_unicode_ci is not the right one.

**Review by markmetcalfe — CHANGES_REQUESTED** [review]
@derschatta Can you also change the char set and collation for mariadb too? Then we can close off #107

**Comment by derschatta** [comment]
> @derschatta Can you also change the char set and collation for mariadb too? Then we can close off #107

done

**Review by markmetcalfe — CHANGES_REQUESTED** [review]
Found issues while testing, can merge https://github.com/derschatta/totara-docker-dev/pull/2/files into this to fix them

**Review by markmetcalfe — APPROVED** [review]

**Pull request merged**